cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -824,6 +883,13 @@ protected JsonObject asJson(KafkaConnectorSpec spec, KafkaConnectorConfiguration
         return connectorConfigJson.put("connector.class", spec.getClassName());
     }
 
+    protected Future<KafkaConnectorStatus> getPreviousKafkaConnectorStatus(CrdOperator<KubernetesClient, KafkaConnector, KafkaConnectorList> resourceOperator,
```

Comment:
Didn't we got the resource with its latest status on the beginning? Can't we use that instead of querying the API server again?

Reply:
As far as I saw, Kafka Connector statuses are recreated from scratch at each reconcile.  
We get connector and tasks statuses from Connect API before auto restarting but it is lacking the auto restart status as it is stored in the resource and the status of the resource is null when we call `maybeCreateOrUpdateConnector` function.

Reply:
They are. But on the beginning, there is always a full `KafkaConnector` resource including the status. I'm not sure how easy or hard would it be to pass it around (currently, I think only the `.spec` part is passed around) -> maybe it would not be possible because of the duality with the Mirror Maker 2 connectors.